contexts/script.tsx

```diff
@@ -137,6 +137,13 @@ const ScriptContextProvider: React.FC<ScriptContextProps> = ({children, initialS
 		}
 	}, [tool, connected, scriptContent, formValues, workspace, thread]);
 
+	useEffect(() => {
+		if (forceRun && socket && connected) {
+			socket.emit("run", script, subTool ? subTool : tool.name, formValues, workspace, thread)
```

Comment:
@tylerslaton Let me know how you would think about the approach... this is the best way I can think of to trigger the socket re-mount logic during reconnection

Reply:
Not a bad approach. I wonder if you could call the restart function from the `useChatSocket.tsx` here instead since that'll wire a bunch of stuff up on your behalf. That being said, I think what you have works.

Reply:
Yeah, trying to avoid the fact that when reconnecting socket we just create a new one. I do feel like the proper solution to handle this in the server side where socket should listen on connect event and remount all the listener, but that's much harder because there are a bunch of paramaters to be passed in when mounting the listener.